tests/test_transaction.py

```diff
@@ -1,57 +1,97 @@
+            }
+        ],
+        "AttributeDefinitions": [
+            {
+                "AttributeName": "MockHash",
+                "AttributeType": "N"
+            },
+            {
+                "AttributeName": "MockRange",
+                "AttributeType": "N"
+            }
+        ]
+    }
+}
 
 
 class TestTransaction:
 
-    def test_commit__not_implemented(self, mocker):
-        t = Transaction(connection=mocker.MagicMock())
+    def test_commit__not_implemented(self):
+        t = Transaction(connection=Connection())
         with pytest.raises(NotImplementedError):
             t._commit()
 
 
 class TestTransactGet:
 
-    def setup(self):
-        self.mock_model_cls = MagicMock(__name__='MockModel')
-        self.mock_model_cls.get_operation_kwargs_from_class.return_value = {}
-
     def test_commit(self, mocker):
-        mock_connection = mocker.MagicMock(spec=Connection)
-        mock_connection.transact_get_items.return_value = {
-            'Responses': [{'Item': {}}]
-        }
+        connection = Connection()
+        mock_connection_transact_get = mocker.patch.object(connection, 'transact_get_items')
 
-        with TransactGet(connection=mock_connection) as t:
-            t.get(self.mock_model_cls, 1, 2)
+        with patch(PATCH_METHOD) as req:
+            req.return_value = MOCK_TABLE_DESCRIPTOR
+            with TransactGet(connection=connection) as t:
+                t.get(MockModel, 1, 2)
 
-        mock_connection.transact_get_items.assert_called_once_with(get_items=[{}], return_consumed_capacity=None)
+        mock_connection_transact_get.assert_called_once_with(
+            get_items=[{'Key': {'MockHash': {'N': '1'}, 'MockRange': {'N': '2'}}, 'TableName': 'mock'}],
+            return_consumed_capacity=None
+        )
 
 
 class TestTransactWrite:
 
     def test_condition_check__no_condition(self):
         with pytest.raises(TypeError):
-            with TransactWrite(connection=MagicMock()) as transaction:
-                transaction.condition_check(MagicMock(__name__='MockModel'), hash_key=1, condition=None)
+            with TransactWrite(connection=Connection()) as transaction:
+                transaction.condition_check(MockModel, hash_key=1, condition=None)
 
     def test_commit(self, mocker):
-        mock_connection = mocker.MagicMock(spec=Connection)
-        with TransactWrite(connection=mock_connection) as t:
-            t._condition_check_items = [{}]
-            t._delete_items = [{}]
-            t._put_items = [{}]
-            t._update_items = [{}]
-
-        mock_connection.transact_write_items.assert_called_once_with(
+        connection = Connection()
+        mock_connection_transact_write = mocker.patch.object(connection, 'transact_write_items')
+        with patch(PATCH_METHOD) as req:
+            req.return_value = MOCK_TABLE_DESCRIPTOR
+            with TransactWrite(connection=connection) as t:
+                t._condition_check_items = [{}]
```

Comment:
Could you use the real methods on `TransactWrite` to add values instead? Would be nice to have a unit test also exercise through the operations kwargs building logic

Reply:
okidokies :0

Reply:
added!